### PR TITLE
DSL syntax

### LINKED ISSUE #56: As a framework user I would higly appreciate a handy DSL to perform these operations
**Initial estimate of effort:** 40

#### Sprint tasks
- [x] #57
- [x] #58
- [x] #79 
- [x] #80
- [x] #85
- [x] #108
- [x] #109

### LINKED ISSUE #109: Add DSL methods (world -= entity;world += system;world -= system)

### PR DESCRIPTION
**Pull request by vitlinda**
Closes #109
Closes #56

### CODE REVIEW

File: core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package dev.atedeg.ecscala.fixtures
+
+import dev.atedeg.ecscala.{ &:, CNil, SystemBuilder, World }
+import dev.atedeg.ecscala.util.types.given
+
+//import scala.language.implicitConversions
```
Comment (ndido98) [inline core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala:6]: Remove

**Review by ndido98 — APPROVED** [review]

**Review by nicolasfara — APPROVED** [review]

File: core/src/main/scala/dev/atedeg/ecscala/dsl/Conversions.scala [hunk 2]
```diff
@@ -8,15 +8,15 @@ trait Conversions {
   /**
    * This conversion enable the removal of a single component from an entity with the following syntax:
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/Conversions.scala:9]:
```suggestion
   * This conversion enables the removal of a single component from an entity with the following syntax:
```

File: core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala [hunk 3]
```diff
@@ -17,46 +17,48 @@ import dev.atedeg.ecscala.{ CList, CNil, Component, Deletable, DeltaTime, Entity
...
  *     * world hasA system[MyComponent &: CNil] { (_,_,_) => {}}
- *     * world hasA system(MySistem())
+ *     * world hasA system(mySistem)
+ *     * world += mySystem
  * }}}
  *
  * '''Remove a System from a World'''
  * {{{
- *   remove MySystem() from world
+ *   * remove mySystem from world
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala:60]: Are you sure this code compiles even if you have no brackets around `mySystem`?

File: core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala [hunk 4]
```diff
@@ -199,11 +214,11 @@ class ECScalaDSLTest extends AnyWordSpec with Matchers with ECScalaDSL {
   }
 
   "clearAll from world" should {
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/dsl/ECScalaDSLTest.scala:216]:
```suggestion
  "clearAllEntities from world" should {
```

File: core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+trait SystemFixture extends ECScalaDSL {
+  lazy val world = World()
+  val entity1 = world hasAn entity withComponent Position(1, 1)
+
+  lazy val mySystem1 = SystemBuilder[Position &: CNil].withBefore { (_, _, _) => () }.withAfter { (_, _, _) =>
+    ()
+  }.withUpdate { (_, c, _) =>
+    val Position(x, y) &: CNil = c
+    Position(x + 3, y + 3) &: CNil
+  }
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala:16]: If you are not specifying any before and after you can simply implement a System with:
```scala
System[Position &: CNil] { (_, components, _) =>
  val Position(x, y) &: CNil = c
  Position(x + 3, y + 3) &: CNil
}
```
It is way more readable

File: core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+  lazy val mySystem2 = SystemBuilder[Position &: CNil].withBefore { (_, _, _) => () }.withAfter { (_, _, _) =>
+    ()
+  }.withUpdate { (_, c, _) =>
+    val Position(x, y) &: CNil = c
+    Position(x + 1, y + 1) &: CNil
+  }
+}
```
Comment (giacomocavalieri) [inline core/src/test/scala/dev/atedeg/ecscala/fixtures/SystemFixture.scala:24]: Same as above

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

File: core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala [hunk 7]
```diff
@@ -17,46 +17,48 @@ import dev.atedeg.ecscala.{ CList, CNil, Component, Deletable, DeltaTime, Entity
...
  *     * world hasA system[MyComponent &: CNil] { (_,_,_) => {}}
- *     * world hasA system(MySistem())
+ *     * world hasA system(mySistem)
+ *     * world += mySystem
  * }}}
  *
  * '''Remove a System from a World'''
  * {{{
- *   remove MySystem() from world
+ *   * remove (mySystem) from world
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala:60]: We used `{}` in all other single-parameter examples, here you are using `()`, pick one and stick with it

File: core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala [hunk 8]
```diff
@@ -95,7 +97,17 @@ trait ECScalaDSL extends ExtensionMethods with Conversions with Syntax {
   /**
    * Keyword that enables the use of the word "remove" in the dsl.
    */
-  def remove[L <: CList: CListTag](componentsList: L): From[Entity, Unit] = FromEntity(componentsList)
+  def remove[L <: CList: CListTag](componentsList: L): From[Entity, Unit] = ComponentsFromEntity(componentsList)
+
+  /**
+   * Keyword that enables the use of the word "remove" in the dsl.
+   */
```
Comment (giacomocavalieri) [inline core/src/main/scala/dev/atedeg/ecscala/dsl/ECScalaDSL.scala:104]: In all this `remove` methods I would shortly describe what is removed from what

**Review by giacomocavalieri — CHANGES_REQUESTED** [review]

**Review by giacomocavalieri — APPROVED** [review]

**Pull request merged**